etna/records/blocks.py

```diff
@@ -84,8 +85,8 @@ def value_from_form(self, value):
             return value
 
         try:
-            return self.target_model.search.get(iaid=value)
-        except (KongAPIError, SearchManagerException):
+            return self.target_model.api.fetch(iaid=value)
+        except (KongAPIError, APIManagerException):
             # If there's a connection issue with Kong, return a stub Record
             # so we have something to render on the ResultsPage edit form.
             return self.target_model(iaid=value)
```

Comment:
- Why do we return a stub record to render something when its an error. Should we not have an error page so that it must be rectified - may be some alert?
- Should these error be handled separately?

Reply:
I believe returning a stub record, may hide the actual error, in this case, we may have two errors KongAPIError or APIManagerException

Reply:
Thanks both.

You're right, We are hiding API errors from the user. 

Let's say record `C123456` is added to an `InsightsPage`. 

But the day after, `C123456` is removed from the catalogue.

If we don't consume the error, editors would be unable to remove the record from the `InsightsPage`.

Reply:
- If the record is removed from the catalogue, then it will also impact its access from the front-end user.  How long does a record live in ES in order for it to be retrieved? 
- Is there some functional understanding between how long a record can be retained in ES, accessed by front-end user and changed by Editor user?

Reply:
Discussed logging exceptions in separate ticket.